pkg/controller/usersignup/usersignup_controller_test.go

```diff
@@ -0,0 +1,404 @@
+	require.NoError(t, err)
+
+	res, err := r.Reconcile(req)
+	require.NoError(t, err)
+	require.Equal(t, reconcile.Result{}, res)
+
+	mur := &v1alpha1.MasterUserRecord{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: userSignup.Name, Namespace: req.Namespace}, mur)
+	require.Error(t, err)
+	require.IsType(t, err, &errors.StatusError{})
+	require.Equal(t, metav1.StatusReasonNotFound, err.(*errors.StatusError).ErrStatus.Reason)
+}
+
+func TestUserSignupWithExistingMUROK(t *testing.T) {
+	userSignup := &v1alpha1.UserSignup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+			Namespace: operatorNamespace,
+			UID: types.UID(uuid.NewV4().String()),
+		},
+		Spec: v1alpha1.UserSignupSpec{
+			UserID: "foo",
+			Approved: false,
+		},
+	}
+
+	// Create a MUR with the same name
+	mur := &v1alpha1.MasterUserRecord{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+			Namespace: operatorNamespace,
+			UID: types.UID(uuid.NewV4().String()),
+		},
+		Spec: v1alpha1.MasterUserRecordSpec{
+			UserID: "foo",
+		},
+	}
+
+	r, req := prepareReconcile(t, userSignup.Spec.UserID, userSignup, mur)
+
+	createMemberCluster(r.client)
+	defer clearMemberClusters(r.client)
+
+	// Create a new ConfigMap and set the user approval policy to automatic
+	cmValues := make(map[string]string)
+	cmValues[config.ToolchainConfigMapUserApprovalPolicy] = config.UserApprovalPolicyAutomatic
+	cm := &v1.ConfigMap{
+		Data: cmValues,
+	}
+	cm.Name = config.ToolchainConfigMapName
+	_, err := r.clientset.CoreV1().ConfigMaps(operatorNamespace).Create(cm)
+	require.NoError(t, err)
+
+	_, err = r.Reconcile(req)
+	require.NoError(t, err)
+
+	key := types.NamespacedName{
+		Namespace: operatorNamespace,
+		Name: "foo",
+	}
+	instance := &v1alpha1.UserSignup{}
+	err = r.client.Get(context.TODO(), key, instance)
+	require.NoError(t, err)
+
+	var cond *v1alpha1.Condition
+	for _, condition := range instance.Status.Conditions {
+		if condition.Type == v1alpha1.UserSignupComplete {
+			cond = &condition
+		}
+	}
+
+	require.NotNil(t, cond)
+	require.Equal(t, v1.ConditionTrue, cond.Status)
+}
+
+func TestUserSignupNoMembersAvailableFails(t *testing.T) {
+	userSignup := &v1alpha1.UserSignup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+			Namespace: operatorNamespace,
+			UID: types.UID(uuid.NewV4().String()),
+		},
+		Spec: v1alpha1.UserSignupSpec{
+			UserID: "foo",
+			Approved: true,
+		},
+	}
+
+	r, req := prepareReconcile(t, userSignup.Spec.UserID, userSignup)
+
+	// Create a new ConfigMap and set the user approval policy to automatic
+	cmValues := make(map[string]string)
+	cmValues[config.ToolchainConfigMapUserApprovalPolicy] = config.UserApprovalPolicyAutomatic
+	cm := &v1.ConfigMap{
+		Data: cmValues,
+	}
+	cm.Name = config.ToolchainConfigMapName
+	_, err := r.clientset.CoreV1().ConfigMaps(operatorNamespace).Create(cm)
+	require.NoError(t, err)
+
+	_, err = r.Reconcile(req)
+	require.Error(t, err)
+	require.IsType(t, SignupError{}, err)
+}
```

Comment:
Let's also add tests for:

- Signup manually approved (it's created as notapproved then updated to approved)
- Mock errors for status update failure, MUR creation failure, Config reading failure. You can test them by mocking the Create/Update/Delete functions in the client. See an example here: https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/useraccount/useraccount_controller_test.go#L251